rllib/core/rl_trainer/tf/tf_rl_trainer.py

```diff
@@ -0,0 +1,196 @@
+import numpy as np
+from typing import Any, Mapping, Tuple, Union, Type
+from ray.rllib.core.optim.rl_optimizer import RLOptimizer
+from ray.rllib.core.optim.marl_optimizer import MultiAgentRLOptimizer
+from ray.rllib.core.rl_trainer.rl_trainer import RLTrainer
+from ray.rllib.core.rl_module.rl_module import RLModule, ModuleID
+from ray.rllib.core.rl_module.marl_module import MultiAgentRLModule
+from ray.rllib.policy.sample_batch import MultiAgentBatch
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.framework import try_import_tf
+from ray.rllib.utils.typing import TensorType
+from ray.rllib.utils.nested_dict import NestedDict
+from ray.rllib.utils.numpy import convert_to_numpy
+
+
+tf1, tf, tfv = try_import_tf()
+tf1.enable_eager_execution()
+
+
+class TfRLTrainer(RLTrainer):
+    """Base class for RLlib TF algorithm trainers."""
```

Comment:
I might remove this docstring so that we inherit the RLTrainer docstring, since its going to be the same